### PR TITLE
[codemod] Add a new utility to rename imports

### PR DESCRIPTION
**Pull request by flaviendelangle**
Needed for #14913
We start to have a lot of codemods that just move imports around, a clean abstraction starts to be worth it

I did not update the grid codemod because their import management differs (there is no nested imports).

Several tree view codemod were not tested because I forgot to add `.test` in the test file, this is why the testing files might change slightly.

### CODE REVIEW

File: packages/x-codemod/src/util/renameImports.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  const importDeclarations = root
+    // Find all the import declarations (import { ... } from '...')
+    .find(j.ImportDeclaration);
+
+  // Rename the nested imports specifiers
+  // - import { A } from '@mui/x-date-pickers/A'
+  // + import { B } from '@mui/x-date-pickers/A'
+  const nestedImportRegExp = new RegExp(`^(${parameters.packageNames.join('|')})/(.*)$`);
+  importDeclarations
+    // Filter out the declarations that are not nested endpoint of the matching packages or that don't have any update to apply
```
Comment (noraleonte) [inline packages/x-codemod/src/util/renameImports.ts:70]:
```suggestion
    // Filter out the declarations that are not nested endpoints of the matching packages or that don't have any update to apply
```

File: packages/x-codemod/src/util/renameImports.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        path.node.imported.name
+      ];
+      return j.importSpecifier(j.identifier(newName));
+    });
+
+  // Rename the nested import declarations
+  // - import { B } from '@mui/x-date-pickers/A'
+  // + import { B } from '@mui/x-date-pickers/B'
+  importDeclarations
+    // Filter out the declarations that are not nested endpoint of the matching packages or that don't have any update to apply
```
Comment (noraleonte) [inline packages/x-codemod/src/util/renameImports.ts:121]:
```suggestion
    // Filter out the declarations that are not nested endpoints of the matching packages or that don't have any update to apply
```

**Review by noraleonte — APPROVED** [review]
Nice addition, super helpful 👌 I left the tiniest nitpick, but LGTM otherwise 🙈

**Review by LukasTy — APPROVED** [review]
LGTM. 👍 
I've added a `bug` label as it looks like it fixes some existing TreeView renaming codemods, doesn't it? 🤔

**Comment by flaviendelangle** [comment]
> I've added a bug label as it looks like it fixes some existing TreeView renaming codemods, doesn't it? 🤔

No, the test were not running and the test output were not correct
The logic was good :+1:

**Pull request merged**